Malformed subscoped credentials error with external s3

### Describe the bug

I'm trying to set up a polaris catalog that points to our on-premises Pure FlashBlade s3 storage instance.

I'm getting the following error when I try to create a table via spark sql shell:

`org.apache.iceberg.exceptions.RESTException: Unable to process: Failed to get subscoped credentials: (Service: Sts, Status Code: 400, Request ID: null) (SDK Attempt Count: 1)`

This is similar to the errors reported in #1146 , but those errors have additional details as to the actual issue after "subscoped credentials:", whereas in this case, the error message is left incomplete.

I believe from https://github.com/apache/polaris/pull/1913 that external s3 providers should be supported currently.

The full stack trace is as follows:

```
spark-sql ()> CREATE NAMESPACE ICE_NS;
Time taken: 1.11 seconds
spark-sql ()> USE NAMESPACE ICE_NS;
Time taken: 0.071 seconds
spark-sql (ICE_NS)> CREATE TABLE PEOPLE (id int, name string) USING iceberg;
25/07/29 17:57:42 ERROR SparkSQLDriver: Failed in [CREATE TABLE PEOPLE (id int, name string) USING iceberg]
org.apache.iceberg.exceptions.RESTException: Unable to process: Failed to get subscoped credentials: (Service: Sts, Status Code: 400, Request ID: null) (SDK Attempt Count: 1)
	at org.apache.iceberg.rest.ErrorHandlers$DefaultErrorHandler.accept(ErrorHandlers.java:248)
	at org.apache.iceberg.rest.ErrorHandlers$TableErrorHandler.accept(ErrorHandlers.java:123)
	at org.apache.iceberg.rest.ErrorHandlers$TableErrorHandler.accept(ErrorHandlers.java:107)
	at org.apache.iceberg.rest.HTTPClient.throwFailure(HTTPClient.java:215)
	at org.apache.iceberg.rest.HTTPClient.execute(HTTPClient.java:299)
	at org.apache.iceberg.rest.BaseHTTPClient.post(BaseHTTPClient.java:88)
	at org.apache.iceberg.rest.RESTSessionCatalog$Builder.create(RESTSessionCatalog.java:771)
	at org.apache.iceberg.CachingCatalog$CachingTableBuilder.lambda$create$0(CachingCatalog.java:264)
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2406)
	at java.base/java.util.concurrent.ConcurrentHashMap.compute(Unknown Source)
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2404)
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2387)
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:108)
	at org.apache.iceberg.shaded.com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62)
	at org.apache.iceberg.CachingCatalog$CachingTableBuilder.create(CachingCatalog.java:260)
	at org.apache.iceberg.spark.SparkCatalog.createTable(SparkCatalog.java:246)
	at org.apache.polaris.spark.SparkCatalog.createTable(SparkCatalog.java:153)
	at org.apache.spark.sql.connector.catalog.TableCatalog.createTable(TableCatalog.java:223)
	at org.apache.spark.sql.execution.datasources.v2.CreateTableExec.run(CreateTableExec.scala:44)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result$lzycompute(V2CommandExec.scala:43)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result(V2CommandExec.scala:43)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.executeCollect(V2CommandExec.scala:49)
	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.$anonfun$applyOrElse$1(QueryExecution.scala:107)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:107)
	at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:98)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:461)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(origin.scala:76)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:461)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.org$apache$spark$sql$catalyst$plans$logical$AnalysisHelper$$super$transformDownWithPruning(LogicalPlan.scala:32)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning(AnalysisHelper.scala:267)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.transformDownWithPruning$(AnalysisHelper.scala:263)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:32)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.transformDownWithPruning(LogicalPlan.scala:32)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformDown(TreeNode.scala:437)
	at org.apache.spark.sql.execution.QueryExecution.eagerlyExecuteCommands(QueryExecution.scala:98)
	at org.apache.spark.sql.execution.QueryExecution.commandExecuted$lzycompute(QueryExecution.scala:85)
	at org.apache.spark.sql.execution.QueryExecution.commandExecuted(QueryExecution.scala:83)
	at org.apache.spark.sql.Dataset.<init>(Dataset.scala:220)
	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:100)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:97)
	at org.apache.spark.sql.SparkSession.$anonfun$sql$4(SparkSession.scala:691)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:682)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:713)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:744)
	at org.apache.spark.sql.SQLContext.sql(SQLContext.scala:651)
	at org.apache.spark.sql.hive.thriftserver.SparkSQLDriver.run(SparkSQLDriver.scala:68)
	at org.apache.spark.sql.hive.thriftserver.SparkSQLCLIDriver.processCmd(SparkSQLCLIDriver.scala:501)
	at org.apache.spark.sql.hive.thriftserver.SparkSQLCLIDriver.$anonfun$processLine$1(SparkSQLCLIDriver.scala:619)
	at org.apache.spark.sql.hive.thriftserver.SparkSQLCLIDriver.$anonfun$processLine$1$adapted(SparkSQLCLIDriver.scala:613)
	at scala.collection.Iterator.foreach(Iterator.scala:943)
	at scala.collection.Iterator.foreach$(Iterator.scala:943)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1431)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at org.apache.spark.sql.hive.thriftserver.SparkSQLCLIDriver.processLine(SparkSQLCLIDriver.scala:613)
	at org.apache.spark.sql.hive.thriftserver.SparkSQLCLIDriver$.main(SparkSQLCLIDriver.scala:310)
	at org.apache.spark.sql.hive.thriftserver.SparkSQLCLIDriver.main(SparkSQLCLIDriver.scala)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:1034)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:199)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:222)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1125)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1134)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)


```
### To Reproduce

I have a docker compose file to initialize the polaris server:
```

services:

  polaris:
    image: apache/polaris:latest
    platform: linux/amd64
    ports:
      - "8181:8181"
      - "8182:8182"
    environment:
      AWS_ACCESS_KEY_ID: <pure-s3-access-key>
      AWS_SECRET_ACCESS_KEY: <pure-s3-secret-key>
      AWS_REGION: us-east-2
      AWS_ENDPOINT_URL_S3: <pure-s3-endpoint-url>
      AWS_ENDPOINT_URL_STS: <same pure s3 endpoint url as immediately above>
      POLARIS_BOOTSTRAP_CREDENTIALS: default-realm,root,secret
      # polaris.features."SUPPORTED_CATALOG_STORAGE_TYPES": "[\"FILE\",\"S3\",\"GCS\",\"AZURE\"]"
      polaris.features.DROP_WITH_PURGE_ENABLED: true # allow dropping tables from the SQL client
      polaris.realm-context.realms: default-realm
      polaris.features."SKIP_CREDENTIAL_SUBSCOPING_INDIRECTION": false
      polaris.features."SUPPORTED_CATALOG_STORAGE_TYPES": "[\"S3\"]" 

```

FYI, I get the same malformed error without the skip credential subscoping indirection = false, and I get the error when I set the docker image version to 1.0.1-incubating-rc0 manually (I've also tried this on 1.0.0-incubating)

I then have two scripts to initialize the polaris server:
```

ACCESS_TOKEN=$(curl -X POST \
  http://localhost:8181/api/catalog/v1/oauth/tokens \
  -d 'grant_type=client_credentials&client_id=root&client_secret=secret&scope=PRINCIPAL_ROLE:ALL' \
  | jq -r '.access_token')

curl -i -X POST \
  -H "Authorization: Bearer $ACCESS_TOKEN" \
  http://localhost:8181/api/management/v1/catalogs \
  -H "Content-Type: application/json" \
  --data '{
    "name": "polariscatalog",
    "type": "INTERNAL",
    "properties": {
      "default-base-location": "s3://polaris-txl25-1",
      "s3.endpoint": "<pure-s3-endpoint>",
      "s3.path-style-access": "true",
      "s3.access-key-id": "<pure-s3-access-key>",
      "s3.secret-access-key": "<pure-s3-secret-key>",
      "s3.region": "us-east-2"
    },
    "storageConfigInfo": {
      "roleArn": "arn:aws:iam::000000000000:role/dummy-polaris-role",
      "storageType": "S3",
      "allowedLocations": [
        "s3://polaris-txl25-1/*"
      ]
    }
  }'
```


and

```
ACCESS_TOKEN=$(curl -X POST \
  http://localhost:8181/api/catalog/v1/oauth/tokens \
  -d 'grant_type=client_credentials&client_id=root&client_secret=secret&scope=PRINCIPAL_ROLE:ALL' \
  | jq -r '.access_token')

# Create a catalog admin role
curl -X PUT http://localhost:8181/api/management/v1/catalogs/polariscatalog/catalog-roles/catalog_admin/grants \
  -H "Authorization: Bearer $ACCESS_TOKEN" \
  -H "Content-Type: application/json" \
  --data '{"grant":{"type":"catalog", "privilege":"CATALOG_MANAGE_CONTENT"}}'

# Create a data engineer role
curl -X POST http://localhost:8181/api/management/v1/principal-roles \
  -H "Authorization: Bearer $ACCESS_TOKEN" \
  -H "Content-Type: application/json" \
  --data '{"principalRole":{"name":"data_engineer"}}'

# Connect the roles
curl -X PUT http://localhost:8181/api/management/v1/principal-roles/data_engineer/catalog-roles/polariscatalog \
  -H "Authorization: Bearer $ACCESS_TOKEN" \
  -H "Content-Type: application/json" \
  --data '{"catalogRole":{"name":"catalog_admin"}}'

# Give root the data engineer role
curl -X PUT http://localhost:8181/api/management/v1/principals/root/principal-roles \
  -H "Authorization: Bearer $ACCESS_TOKEN" \
  -H "Content-Type: application/json" \
  --data '{"principalRole": {"name":"data_engineer"}}'
```


These scripts initiate the catalog, and then create a role for the user.

Then, I use the following docker file to create a spark environment that has my spark submit script

```
FROM docker-hub/spark:3.5.6

ENV AWS_ACCESS_KEY_ID <pure-s3-access-key>
ENV AWS_SECRET_ACCESS_KEY <pure-s3-secret-key>
ENV AWS_ENDPOINT_URL <pure-s3-endpoint>
ENV AWS_REGION us-east-2

COPY txl25-polaris-sql.sh /opt/spark/bin/txl25-polaris-sql.sh

USER root
RUN apt-get update
RUN apt-get install libcap2-bin libcap-dev less -y
USER spark
```

And then, from within the running spark docker container (running on host networking so that it can talk to polaris), I execute the following:

```
./spark-sql \
--packages org.apache.polaris:polaris-spark-3.5_2.12:1.0.0-incubating,org.apache.iceberg:iceberg-aws-bundle:1.9.0,io.delta:delta-spark_2.12:3.3.1 \
 --conf spark.driver.extraJavaOptions="-Divy.cache.dir=/tmp -Divy.home=/tmp" \
--conf spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,io.delta.sql.DeltaSparkSessionExtension \
--conf spark.sql.catalog.lorna.warehouse=polariscatalog \
--conf spark.sql.catalog.lorna.header.X-Iceberg-Access-Delegation=vended-credentials \
--conf spark.sql.catalog.lorna=org.apache.polaris.spark.SparkCatalog \
--conf spark.sql.catalog.lorna.uri=http://localhost:8181/api/catalog \
--conf spark.sql.catalog.lorna.credential='root:secret' \
--conf spark.sql.catalog.lorna.scope='PRINCIPAL_ROLE:ALL' \
--conf spark.sql.catalog.lorna.token-refresh-enabled=true \
--conf spark.sql.catalog.lorna.io-impl=org.apache.iceberg.io.ResolvingFileIO \
--conf spark.hadoop.fs.s3.impl=org.apache.hadoop.fs.s3a.S3AFileSystem \
--conf spark.sql.catalog.lorna.s3.region=us-east-2 \
--conf spark.sql.catalog.lorna.s3.endpoint=<pure-s3-endpoint>
```

From within the spark sql shell, I execute the following:
```

USE lorna;
CREATE NAMESPACE ICE_NS;
USE NAMESPACE ICE_NS;
CREATE TABLE PERSON (id int, name string) USING iceberg;
```

And then the error report occurs.

### Actual Behavior

_No response_

### Expected Behavior

I would expect the system to be able to interact with my custom s3 endpoint, but at the very least I think that the error message should be fully formed as to give me a better clue why this is failing.

### Additional context

I do get this warning when selecting my spark catalog, so maybe this is relevant:

```
spark-sql (default)> USE lorna;
25/07/29 18:23:08 WARN AuthManagers: Inferring rest.auth.type=oauth2 since property credential was provided. Please explicitly set rest.auth.type to avoid this warning.
25/07/29 18:23:08 WARN OAuth2Manager: Iceberg REST client is missing the OAuth2 server URI configuration and defaults to http://localhost:8181/api/catalog/v1/oauth/tokens. This automatic fallback will be removed in a future Iceberg release.It is recommended to configure the OAuth2 endpoint using the 'oauth2-server-uri' property to be prepared. This warning will disappear if the OAuth2 endpoint is explicitly configured. See https://github.com/apache/iceberg/issues/10537
25/07/29 18:23:10 WARN ObjectStore: Failed to get database global_temp, returning NoSuchObjectException
Time taken: 2.29 
```

Here's the last bit of my polaris docker logs:

```
2025-07-29 18:47:05,312 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - - [29/Jul/2025:18:47:05 +0000] "POST /api/catalog/v1/oauth/tokens HTTP/1.1" 200 753
2025-07-29 18:47:05,451 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - root [29/Jul/2025:18:47:05 +0000] "GET /api/catalog/v1/config?warehouse=polariscatalog HTTP/1.1" 200 2351
2025-07-29 18:47:05,532 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - root [29/Jul/2025:18:47:05 +0000] "GET /api/catalog/v1/config?warehouse=polariscatalog HTTP/1.1" 200 2351
2025-07-29 18:47:15,943 INFO  [org.apa.pol.ser.exc.IcebergExceptionMapper] [,default-realm] [,,,] (executor-thread-1) Handling runtimeException Namespace does not exist: ice_ns
2025-07-29 18:47:15,961 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - root [29/Jul/2025:18:47:15 +0000] "GET /api/catalog/v1/polariscatalog/namespaces/ice_ns HTTP/1.1" 404 101
2025-07-29 18:47:16,066 INFO  [org.apa.pol.ser.cat.ice.IcebergCatalogHandler] [,default-realm] [,,,] (executor-thread-1) Initializing non-federated catalog
2025-07-29 18:47:16,109 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - root [29/Jul/2025:18:47:16 +0000] "POST /api/catalog/v1/polariscatalog/namespaces HTTP/1.1" 200 96
2025-07-29 18:48:00,501 INFO  [org.apa.pol.ser.cat.ice.IcebergCatalogHandler] [,default-realm] [,,,] (executor-thread-1) Initializing non-federated catalog
2025-07-29 18:48:00,525 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - root [29/Jul/2025:18:48:00 +0000] "GET /api/catalog/v1/polariscatalog/namespaces?pageToken= HTTP/1.1" 200 50
2025-07-29 18:48:06,053 INFO  [org.apa.pol.ser.cat.ice.IcebergCatalogHandler] [,default-realm] [,,,] (executor-thread-1) Initializing non-federated catalog
2025-07-29 18:48:06,060 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - root [29/Jul/2025:18:48:06 +0000] "GET /api/catalog/v1/polariscatalog/namespaces/ice_ns HTTP/1.1" 200 96
2025-07-29 18:48:26,456 INFO  [org.apa.pol.ser.exc.IcebergExceptionMapper] [,default-realm] [,,,] (executor-thread-1) Handling runtimeException Table does not exist: ice_ns.person
2025-07-29 18:48:26,459 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-1) 172.19.0.1 - root [29/Jul/2025:18:48:26 +0000] "GET /api/catalog/v1/polariscatalog/namespaces/ice_ns/tables/person?snapshots=all HTTP/1.1" 404 100
2025-07-29 18:48:26,485 INFO  [org.apa.pol.ser.exc.IcebergExceptionMapper] [,default-realm] [,,,] (executor-thread-4) Handling runtimeException Generic table does not exist: ice_ns.person
2025-07-29 18:48:26,487 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-4) 172.19.0.1 - root [29/Jul/2025:18:48:26 +0000] "GET /api/catalog/polaris/v1/polariscatalog/namespaces/ice_ns/generic-tables/person HTTP/1.1" 404 108
2025-07-29 18:48:26,614 INFO  [org.apa.pol.ser.cat.ice.IcebergCatalogHandler] [,default-realm] [,,,] (executor-thread-4) Initializing non-federated catalog
2025-07-29 18:48:26,640 INFO  [org.apa.ice.BaseMetastoreCatalog] [,default-realm] [,,,] (executor-thread-4) Table properties set at catalog level through catalog properties: {}
2025-07-29 18:48:26,650 INFO  [org.apa.ice.BaseMetastoreCatalog] [,default-realm] [,,,] (executor-thread-4) Table properties enforced at catalog level through catalog properties: {}
2025-07-29 18:48:27,087 INFO  [org.apa.pol.ser.exc.IcebergExceptionMapper] [,default-realm] [,,,] (executor-thread-4) Handling runtimeException Failed to get subscoped credentials: (Service: Sts, Status Code: 400, Request ID: null) (SDK Attempt Count: 1)
2025-07-29 18:48:27,088 INFO  [io.qua.htt.access-log] [,default-realm] [,,,] (executor-thread-4) 172.19.0.1 - root [29/Jul/2025:18:48:27 +0000] "POST /api/catalog/v1/polariscatalog/namespaces/ice_ns/tables HTTP/1.1" 422 183

```

### System information

Tested with polaris docker images 1.0.1-incubating-rc0 and 1.0.0-incubating. Using a spark docker image versioned 3.5.6

@dimas-b  / @adutra : I attempted to create a catalog and tables using Polaris Catalog service version 1.2.0, but was unable to create a table when using a non-AWS S3-compatible object storage service without STS. Below, I have summarized the testing steps and observations. Please let me know if I have overlooked any configuration or setup requirements.

Aim - To create a catalog and table in Polaris Catalog using a non-AWS S3-compatible storage endpoint.

Setup Details

1. Polaris Catalog Service
	- Version: 1.2.0 (Docker image)
	- Application property file added before service start.
	- Environment variable AWS_SHARED_CREDENTIALS_FILE set to a credentials file containing non-AWS S3 access key and secret.
	- Expectation: The default credential provider chain in Polaris should use these credentials.

2. Python Application / REST API 
	- Used to call Polaris REST API for catalog creation and to set up roles, permissions, and related configurations.

3. Spark Client
	- Used to create a table in the catalog.
	- X-Iceberg-Access-Delegation property is not set in spark client 


Polaris Log Snippet (Catalog Creation)

Created new catalog class PolarisCatalog {
    class Catalog {
        type: INTERNAL
        name: demo_catalog
        properties: class CatalogProperties {
            {default-base-location=s3://non-aws-bucket/demo_catalog/warehouse/}
            defaultBaseLocation: s3://non-aws-bucket/demo_catalog/warehouse/
        }
        createTimestamp: 1234567891011
        lastUpdateTimestamp: 0
        entityVersion: 1
        storageConfigInfo: class AwsStorageConfigInfo {
            class StorageConfigInfo {
                storageType: S3
                allowedLocations: [s3://non-aws-bucket/demo_catalog/warehouse/*, s3://non-aws-bucket/demo_catalog/warehouse/]
            }
            roleArn: null
            externalId: null
            userArn: null
            region: null
            endpoint: https://s3.object.net:port
            stsEndpoint: null
            stsUnavailable: true
            endpointInternal: https://s3.object.net:port
            pathStyleAccess: true
        }
    }
}

Observed Error - 
Caused by: java.net.UnknownHostException: non-aws-bucket-name.s3.region.amazonaws.com: Name or service not known


Questions & Clarifications Needed

1. Unexpected AWS Hostname Resolution
	- Why is Polaris still attempting to connect to non-aws-bucket-name.s3.region.amazonaws.com (AWS domain host) instead of the specified non-AWS S3 endpoint (https://s3.object.net:port)?
	- Is there a configuration or property that needs to be set to ensure Polaris uses the custom endpoint for S3 operations?

2. Credential Provider Chain Behavior
	- When stsUnavailable is set to true, does Polaris use the default credential provider chain to retrieve access keys and secrets from the environment variable (AWS_SHARED_CREDENTIALS_FILE)?
	- Are there additional steps required to ensure Polaris picks up credentials for non-AWS S3 endpoints?
	
Due to time constraints, I have not been able to review the Polaris source code in detail.

--- a/site/content/in-dev/unreleased/getting-started/creating-a-catalog/s3/catalog-ozone.md
+++ b/site/content/in-dev/unreleased/getting-started/creating-a-catalog/s3/catalog-ozone.md
@@ -1,0 +1,69 @@
+---
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+title: Creating a catalog on Apache Ozone
+linkTitle: Ozone
+type: docs
+weight: 150
+---
+
+When creating a catalog based on [Apache Ozone](https://ozone.apache.org/) storage it is important to 
+configure the `endpoint` property to point to your own storage cluster. If the `endpoint` property is
+not set, Polaris will attempt to contact AWS storage services (which is certain to fail in this case).
+
+Note: the `--no-sts` CLI option instructs Polaris to avoid calling STS endpoints with `AssumeRoles`
+requests. This means that both Polaris and its clients (engines) will have to use distinct (local)
+credentials for accessing storage. Engines should _not_ request credential vending in this case
+(should not use the `X-Iceberg-Access-Delegation=vended-credentials` header).
+
+Note: the region setting is not required by Ozone, but it is set in this example for the sake of
+simplicity as it is usually required by the AWS SDK (used internally by Polaris). One can also 
+set the `AWS_REGION` environment variable in the Polaris server process and avoid setting region
+as a catalog property.
+
+Note: the name `quickstart_catalog` from the example below is referenced in other Getting Started 
+examples, but of course, it can be any valid catalog name.
+
+```shell
+CLIENT_ID=root
+CLIENT_SECRET=s3cr3t
+DEFAULT_BASE_LOCATION=s3://example-bucket/my_data
+
+./polaris \
+  --client-id ${CLIENT_ID} \
+  --client-secret ${CLIENT_SECRET} \
+  catalogs \
+  create \
+  --storage-type s3 \
+  --endpoint http://127.0.0.1:9878
+  --no-sts
+  --path-style-access
+  --default-base-location ${DEFAULT_BASE_LOCATION} \
+  --region irrelevant \
+  quickstart_catalog
+```
+
+In more complex deployments it may be necessary to configure different endpoints for S3 requests
+and for STS (AssumeRole) requests. This can be achieved via the `--sts-endpoint` CLI option.
+
+Additionally, the `--endpoint-internal` CLI option cane be used to set the S3 endpoint for use by
+the Polaris Server itself, if it needs to be different from the endpoint used by clients / engines.
+
+A usable Apache Ozone example for `docker-compose` is available in the Polaris source code under the
+[getting-started/ozone](https://github.com/apache/polaris/tree/main/getting-started/ozone) module.

